packages/react-form/src/hooks/field/test/field.test.tsx

```diff
@@ -191,6 +191,100 @@ describe('useField', () => {
+          const text = 'Test field';
+
+          return (
+            <>
+              <label htmlFor="test-field">
+                {text}
+                <input
+                  id="test-field"
+                  name="test-field"
+                  value={field.value}
+                  onChange={field.onChange}
+                  onBlur={field.onBlur}
+                />
+              </label>
+              {field.allErrors && <p>{field.allErrors}</p>}
+            </>
+          );
+        }
+
+        const firstFailingValidator = (value: string) =>
+          `${value} tastes most foul`;
+        const secondFailingValidator = (value: string) => `${value} is horrid`;
+
+        const fieldConfig = {
+          value: 'old title',
+          validates: [
+            alwaysPass,
+            firstFailingValidator,
+            alwaysPass,
+            secondFailingValidator,
+          ],
+        };
+        const wrapper = mount(<TestField config={fieldConfig} />);
+        const newValue = faker.commerce.product();
+
+        wrapper.find('input')!.trigger('onChange', changeEvent(newValue));
+        wrapper.find('input')!.trigger('onBlur', blurEvent());
+
+        expect(wrapper).toContainReactComponent('p', {
+          children: [
+            firstFailingValidator(newValue),
+            secondFailingValidator(newValue),
+          ],
+        });
+      });
+
+      it('only updates the allErrors property if validation errors have changed', () => {
+        function TestField({config}: {config: string | FieldConfig<string>}) {
+          const field = useField(config);
+          const text = 'Test field';
+
+          return (
+            <>
+              <label htmlFor="test-field">
+                {text}
+                <input
+                  id="test-field"
+                  name="test-field"
+                  value={field.value}
+                  onChange={field.onChange}
+                  onBlur={field.onBlur}
+                />
+              </label>
+              {field.allErrors && <p>{field.allErrors}</p>}
+            </>
+          );
+        }
+
+        const failingValidator = (value: string) => `${value} tastes most foul`;
+
+        const fieldConfig = {
+          value: 'old title',
+          validates: [failingValidator],
+        };
+
+        const wrapper = mount(<TestField config={fieldConfig} />);
+        const newValue = faker.commerce.product();
+
+        wrapper.find('input')!.trigger('onChange', changeEvent(newValue));
+        wrapper.find('input')!.trigger('onBlur', blurEvent());
+
+        const allErrorsFirstValidation = wrapper.find('p').props.children;
+
+        wrapper.find('input')!.trigger('onChange', changeEvent(newValue));
+        wrapper.find('input')!.trigger('onBlur', blurEvent());
+
+        const allErrorsSecondValidation = wrapper.find('p').props.children;
+
+        expect(allErrorsFirstValidation).toBe(allErrorsSecondValidation);
```

Comment:
Nice!

* I'd like to change the expect from `tobe` to `toStrictEqual`, as that would be closer to the quality checks in `useEffect`'s deps list.
* Also, can we have a second test for the happy-path, essentially the same, but testing the equality check when `allErrorsFirstValidation === allErrorsSecondValidation` when both arrays have zero elements?

Reply:
Done :)